pkg/backend/filestate/backend.go

```diff
@@ -71,18 +84,41 @@ func (r localBackendReference) Name() tokens.QName {
 	return r.name
 }
 
-func IsLocalBackendURL(url string) bool {
-	return strings.HasPrefix(url, localBackendURLPrefix)
+func IsFileStateBackendURL(url string) bool {
+	for _, scheme := range supportedSchemes {
+		if strings.HasPrefix(url, scheme) {
+			return true
+		}
+	}
+	return false
 }
 
 func New(d diag.Sink, url, stackConfigFile string) (Backend, error) {
-	if !IsLocalBackendURL(url) {
-		return nil, errors.Errorf("local URL %s has an illegal prefix; expected %s", url, localBackendURLPrefix)
+	if !IsFileStateBackendURL(url) {
+		return nil, errors.Errorf("local URL %s has an illegal prefix; expected one of %s", url, strings.Join(supportedSchemes, ", "))
+	}
+
+	if strings.HasPrefix(url, "file://") {
+		// For file:// backend, ensure a relative path is resolved. fileblob only supports absolute paths.
+		localPath, _ := filepath.Abs(strings.TrimPrefix(url, "file://"))
+		url = path.Join("file://", localPath)
```

Comment:
`path.Join` is not what you want here; it drops duplicate slashes:

https://play.golang.org/p/lgnymhswC54

The OpenBucket example [here](https://godoc.org/gocloud.dev/blob/fileblob) may be helpful.

```
// On Unix, append the dir to "file://".
// On Windows, convert "\" to "/" and add a leading "/":
dirpath := filepath.ToSlash(dir)
if os.PathSeparator != '/' && !strings.HasPrefix(dirpath, "/") {
    dirpath = "/" + dirpath
}

b, err := blob.OpenBucket(ctx, "file://"+dirpath)
...
```

Reply:
path.Join could be used, but this is probably a cheesy way to accomplish this.

https://play.golang.org/p/Ih_7Fqm_wzz

Reply:
https://play.golang.org/p/5wbCMuJpF7W
is closer. But it doesn't do the special Windows things.

Reply:
oh, that's neat!

Reply:
Done.  Thanks!